File: launchdarkly/java/src/main/java/oseg/launchdarkly_examples/PatchHoldoutExample.java
package oseg.launchdarkly_examples;

import com.launchdarkly.client.ApiException;
import com.launchdarkly.client.Configuration;
import com.launchdarkly.client.api.*;
import com.launchdarkly.client.auth.*;
import com.launchdarkly.client.JSON;
import com.launchdarkly.client.model.*;

import java.io.File;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class PatchHoldoutExample
{
    public static void main(String[] args)
    {
        var config = Configuration.getDefaultApiClient();
        ((ApiKeyAuth) config.getAuthentication("ApiKey")).setApiKey("YOUR_API_KEY");

        var holdoutPatchInput = new HoldoutPatchInput();
        holdoutPatchInput.instructions(JSON.deserialize("""
            [
                {
                    "kind": "updateName",
                    "value": "Updated holdout name"
                }
            ]
        """, List.class));
        holdoutPatchInput.comment("Optional comment describing the update");

        try
        {
            var response = new HoldoutsBetaApi(config).patchHoldout(
                "projectKey_string", // projectKey
                "environmentKey_string", // environmentKey
                "holdoutKey_string", // holdoutKey
                holdoutPatchInput
            );

            System.out.println(response);
        } catch (ApiException e) {
            System.err.println("Exception when calling HoldoutsBetaApi#patchHoldout");
            System.err.println("Status code: " + e.getCode());
            System.err.println("Reason: " + e.getResponseBody());
            System.err.println("Response headers: " + e.getResponseHeaders());
            e.printStackTrace();
        }
    }
}

File: launchdarkly/kotlin/src/main/kotlin/oseg/launchdarkly_examples/PatchHoldoutExample.kt
package oseg.launchdarkly_examples

import com.launchdarkly.client.infrastructure.*
import com.launchdarkly.client.apis.*
import com.launchdarkly.client.models.*

import java.io.File
import java.time.LocalDate
import java.time.OffsetDateTime
import kotlin.collections.ArrayList
import kotlin.collections.List
import kotlin.collections.Map
import com.squareup.moshi.adapter

@ExperimentalStdlibApi
class PatchHoldoutExample
{
    fun patchHoldout()
    {
        ApiClient.apiKey["Authorization"] = "YOUR_API_KEY"

        val holdoutPatchInput = HoldoutPatchInput(
            instructions = Serializer.moshi.adapter<List<Map<String, Any>>>().fromJson("""
                [
                    {
                        "kind": "updateName",
                        "value": "Updated holdout name"
                    }
                ]
            """)!!,
            comment = "Optional comment describing the update",
        )

        try
        {
            val response = HoldoutsBetaApi().patchHoldout(
                projectKey = "projectKey_string",
                environmentKey = "environmentKey_string",
                holdoutKey = "holdoutKey_string",
                holdoutPatchInput = holdoutPatchInput,
            )

            println(response)
        } catch (e: ClientException) {
            println("4xx response calling HoldoutsBetaApi#patchHoldout")
            e.printStackTrace()
        } catch (e: ServerException) {
            println("5xx response calling HoldoutsBetaApi#patchHoldout")
            e.printStackTrace()
        }
    }
}
